ibc-clients/ics07-tendermint/types/src/client_state.rs

```diff
@@ -176,11 +176,9 @@ impl ClientState {
             });
         }
 
-        // Disallow empty proof-specs
-        if self.proof_specs.is_empty() {
-            return Err(Error::Validation {
-                reason: "ClientState proof-specs cannot be empty".to_string(),
-            });
+        // Sanity checks on client proof specs
+        if let Err(e) = self.proof_specs.validate() {
```

Comment:
Just another suggestion:
Given the added `InvalidProofSpec`, let's add the following in the `error.rs`

```rust
impl From<CommitmentError> for Error {
    fn from(e: CommitmentError) -> Self {
        Self::InvalidProofSpec(e)
    }
}
```

so that can leave with `self.proof_specs.validate()?`

Reply:
8a985086